x/foundation/keeper/internal/treasury.go

```diff
@@ -6,27 +6,16 @@ import (
 	"github.com/line/lbm-sdk/x/foundation"
 )
 
-func (k Keeper) CollectFoundationTax(ctx sdk.Context) error {
-	// fetch and clear the collected fees for the fund, since this is
-	// called in BeginBlock, collected fees will be from the previous block
-	feeCollector := k.authKeeper.GetModuleAccount(ctx, k.feeCollectorName)
-	feesCollectedInt := k.bankKeeper.GetAllBalances(ctx, feeCollector.GetAddress())
+func (k Keeper) CollectFoundationTax(ctx sdk.Context, from sdk.AccAddress) error {
+	feesCollectedInt := k.bankKeeper.GetAllBalances(ctx, from)
```

Comment:
Why do you extract the feeCollectorModule address out of this function?
I think this function doesn't know the `from` is the address of collector module. 
And it seems to open up the ability to insert other addresses as `from` except collection module.

Reply:
It does not allow the app developers to use another address, because the function `CollectFoundationTax()` is not API of this module. So, one must forge this package to do the bad things.
But I don't mind to revert this change, so I will revert it.